### ISSUE TITLE
Error: Could not find adapter type sqlserver

### ISSUE THREAD

**Issue by larsjohanssonblank** [lightdash/lightdash#37]
Docker Desktop on Windows 10.
dbt project and profile.yml working.

Running 
docker run -p "8080" -v "C:\projects\myproject" -v "C:\Users\myuser\.dbt" lightdash/lightdash

(after downloading docker images) gives the result:

_yarn run v1.22.5
$ node dist/index.js
   |     |     |     |     |     |     |
   |     |     |     |     |     |     |
   |     |     |     |     |     |     |
 \ | / \ | / \ | / \ | / \ | / \ | / \ | /
  \|/   \|/   \|/   \|/   \|/   \|/   \|/
------------------------------------------
Launch lightdash at http://localhost:8080
------------------------------------------
  /|\   /|\   /|\   /|\   /|\   /|\   /|\
 / | \ / | \ / | \ / | \ / | \ / | \ / | \
   |     |     |     |     |     |     |
   |     |     |     |     |     |     |
   |     |     |     |     |     |     |
(node:28) UnhandledPromiseRejectionWarning: Error: spawn dbt ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:467:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:28) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:28) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code._

and browsing to localhost:8080 says "This site can’t be reached"

Are there issues with running this on Windows?

**Comment by owlas** [comment]
Hey @larsjohanssonblank - I haven't tested this on Windows yet so thanks for the help!

What shell are you using for running docker? Could you try the following syntax:

```shell
docker run \-p "8080:8080" -v "/c/projects/myproject:/usr/app/dbt" -v "/c/Users/myuser/.dbt:/usr/app/profiles" lightdash/lightdash
```

If that still shows an error, could you try running `dbt debug` and sharing the output:

**Comment by larsjohanssonblank** [comment]
Thanks for the reply.
The UI at localhost:8080 is reachable, but I get a ton of errors/warnings, It may have to do something with the sqlserver adapter we are using.

_Launch lightdash at http://localhost:8080
------------------------------------------
  /|\   /|\   /|\   /|\   /|\   /|\   /|\
 / | \ / | \ / | \ / | \ / | \ / | \ / | \
   |     |     |     |     |     |     |
   |     |     |     |     |     |     |
   |     |     |     |     |     |     |
{"timestamp": "2021-06-04T09:15:15.634612Z", "message": "Running with dbt=0.19.1", "channel": "dbt", "level": 11, "levelname": "INFO", "thread_name": "MainThread", "process": 35, "extra": {"run_state": "internal"}}
{"timestamp": "2021-06-04T09:15:15.655566Z", "message": "Encountered an error while reading profiles:", "channel": "dbt", "level": 14, "levelname": "ERROR", "thread_name": "MainThread", "process": 35, "extra": {"run_state": "internal"}}
{"timestamp": "2021-06-04T09:15:15.655762Z", "message": "  ERROR Runtime Error\n  Credentials in profile \"dbt-dw\", target \"dev\" invalid: Runtime Error\n    **Could not find adapter type sqlserver!"**, "channel": "dbt", "level": 14, "levelname": "ERROR", "thread_name": "MainThread", "process": 35, "extra": {"run_state": "internal"}}_


The dbt debug says everything is ok:
dbt debug
_Running with dbt=0.19.1
dbt version: 0.19.1
python version: 3.7.9
python path: c:\python\python.exe
os info: Windows-10-10.0.18362-SP0
Using profiles.yml file at C:\Users\xxxx\.dbt\profiles.yml
Using dbt_project.yml file at C:\projects\dbt-dw\dbt_project.yml

Configuration:
  profiles.yml file [OK found and valid]
  dbt_project.yml file [OK found and valid]

Required dependencies:
 - git [OK found]

Connection:
  server: mysqlserverxxx
  database: mydbnamexxx
  schema: dbt
  port: 1433
  UID: None
  client_id: None
  authentication: Windows Login
  encrypt: False
  trust_cert: False
  Connection test: OK connection ok_

GUI screenshot:
![image](https://user-images.githubusercontent.com/7777769/120778973-e668f780-c526-11eb-84b6-86b95bb63f17.png)

And everything regarding dbt executions etc works without problems (just to clarify this)

**Comment by owlas** [comment]
That's perfect thanks, looks like we don't have the sql server dbt adapter installed, will open an issue now to fix it now

**Comment by larsjohanssonblank** [comment]
thanks a lot for helping out.
This project looks so promising. 
I like the ecosystem that is evolving round dbt  making it a complete data platform (even for us Windows/SQL Server people ;-))

**Comment by owlas** [comment]
@larsjohanssonblank which driver are you using in your `profiles.yml`?

**Comment by larsjohanssonblank** [comment]
Do you mean this section for our SQL Server?

dbt-dw:
  outputs:
    mydev:
      type: sqlserver
      driver: 'ODBC Driver 17 for SQL Server'
      
      
 and we are using this adapter for SQL Server.
 https://github.com/dbt-msft/dbt-sqlserver

**Comment by owlas** [comment]
Yes that's perfect thanks. I've opened #50 - just need to fix a bug with grabbing the column types from sqlserver

**Comment by owlas** [comment]
@larsjohanssonblank I think this is working from our side on #50 but waiting to resolve a question with the `dbt-sqlserver` project: https://github.com/dbt-msft/dbt-sqlserver/issues/136

**Comment by owlas** [comment]
Hey @larsjohanssonblank - still waiting on a response from https://github.com/dbt-msft/dbt-sqlserver/issues/136

based on activity in the repo I think they're getting to the issue.

However, in the meantime could you see if the following works for you? I've built this PR #50 in to a docker image. Does this work for you now:

`docker pull lightdash/lightdash:876b7c97d85861942ad9ef27a69265d1fa1a1f81`

**Comment by larsjohanssonblank** [comment]
Hi @owlas,

pulled (docker pull lightdash/lightdash:876b7c97d85861942ad9ef27a69265d1fa1a1f81)
and it got a bit longer this time, but it looks like we still have issues with the sqlserver adapter.
Running with 
docker run -p "8080:8080" -v "/c/projects/dbt-dw:/usr/app/dbt" -v "/c/Users/xxx/.dbt:/usr/app/profiles" lightdash/lightdash

(where xxx is my windows user)

![image](https://user-images.githubusercontent.com/7777769/122273894-4033cf00-cee2-11eb-88f2-b50aef29f418.png)


on localhost:8080

![image](https://user-images.githubusercontent.com/7777769/122273555-d7e4ed80-cee1-11eb-8d51-3fa32c23d028.png)

**Comment by owlas** [comment]
Hey @larsjohanssonblank 

I've just been testing with the new SQL Server fix (all seems good).

In playing around with SQL Server, I've actually noticed that our query builder doesn't support SQL Server's syntax. I hadn't even considered this when first investigating this issue.

Right now we have a single sql builder, which works well for dbt's officially supported dbs. But to support anything else we'd need to build our own adapter system. This is 100% on the roadmap.

For now, we'll have to say that we don't officially support SQL Server. I have added an issue to track this and documented the necessary steps.

Is it alright to close this issue in favour of #46 ?

This is something we're aiming to support in the next few weeks.
